core/src/main/java/org/jruby/ext/socket/SocketUtils.java

```diff
@@ -502,7 +501,25 @@ private static InetAddress specialAddress(String addressString) throws UnknownHo
 
     public static IRubyObject getaddress(ThreadContext context, IRubyObject hostname) {
         try {
-            return RubyString.newInternalFromJavaExternal(context.runtime, InetAddress.getByName(hostname.convertToString().toString()).getHostAddress());
+            String hostnameString = hostname.convertToString().toString();
+            InetAddress address = InetAddress.getByName(hostnameString);
+
+            String ipv4MappedAddressPrefix = "::ffff:";
```

Comment:
Can you make these two string static finals?

Reply:
I had to move them outside of the method to make them static.

Reply:
@mullermp yeah this was just to make those strings once.  Thanks!